perses/annihilation/relative.py

```diff
@@ -244,20 +252,31 @@ def __init__(self,
         self._hybrid_system.setDefaultPeriodicBoxVectors(*box_vectors)
         _logger.info(f"getDefaultPeriodicBoxVectors added to hybrid: {box_vectors}")
 
-        #assign atoms to one of the classes described in the class docstring
-        self._atom_classes = self._determine_atom_classes()
-        _logger.info("Determined atom classes.")
-
         #create the opposite atom maps for use in nonbonded force processing; let's omit this from logger
         self._hybrid_to_old_map = {value : key for key, value in self._old_to_hybrid_map.items()}
         self._hybrid_to_new_map = {value : key for key, value in self._new_to_hybrid_map.items()}
 
+        #assign atoms to one of the classes described in the class docstring
+        self._atom_classes = self._determine_atom_classes()
+        _logger.info("Determined atom classes.")
+
         #construct dictionary of exceptions in old and new systems
         _logger.info("Generating old system exceptions dict...")
         self._old_system_exceptions = self._generate_dict_from_exceptions(self._old_system_forces['NonbondedForce'])
         _logger.info("Generating new system exceptions dict...")
         self._new_system_exceptions = self._generate_dict_from_exceptions(self._new_system_forces['NonbondedForce'])
 
+        #conduct a sanity check to make sure that the hybrid maps of the old and new system exception dict keys do not contain both environment and unique_old/new atoms
```

Comment:
ensure that all of the atom flavor sets (i.e. env, core, unique_new, unique_old) are disjoint